### ISSUE TITLE
Responsive text-{size} classes override non-responsive leading classes

### ISSUE THREAD

**Issue by happycollision** [tailwindlabs/tailwindcss#2907]
Dupe of #2852 and #2603 but perhaps the real problem is missing documentation. I’m happy to take a look and come up with something. Probably riffing off of Adam’s comments in #2603. I’ll close this ticket once I open a ticket on the docs repo.

### Describe the problem:

When you set a leading value and a responsive text size, the leading value is ignored.

### Link to a minimal reproduction:

https://play.tailwindcss.com/E6P5LMY1Hl

You'll find a description at the top of the code to reproduce. But, minus the playground, here's what I wrote:

### Repro Steps

Responsive text size classes override leading classes. The workaround is to add your leading class again at the responsive breakpoint you've chosen for your text.

Considering the following code...

```html
<!-- the last class in the next line is intentionally missing a colon -->
<div class="text-base sm:text-lg leading-loose smleading-loose">
  Lots of text so you can see the wrapping.... Etc
</div>
```

1. Ensure that the rendered output is in a wide enough viewport to hit at least the "sm" breakpoint.
2. In the first line of code above, add the missing colon to make the last class in the list read "sm:leading-loose".

You'll observe the text jump into place, even though the leading had already been set to "leading-loose".

Basically, any time you use a responsive prefix with a text size, you lose whatever non-responsive leading you may have already set.

### Thoughts

I don't really know without digging into all your css definitions if this can be mitigated without losing some other important functionality of the `text-{size}` classes. Since they contain `line-height` values, they override the `leading` classes when media queries increase their specificity.

So if this is unfixable, perhaps the documentation should mention it? And maybe the vscode extension can look for it when it is obvious?

**Comment by fedeci** [comment]
Duplicate of #2606

**Comment by happycollision** [comment]
Oops. Actually a dupe of #2852. I always forget that GitHub only searches open issues by default.

So yeah, I’ll change the thoughts above to emphasize documentation as the request here. If everyone is amenable, I’ll take on the work.

**Comment by adamwathan** [comment]
Yeah as discussed previously this is the intended behavior and the solution is to either apply a leading class alongside the responsive font size class, or disable default line heights in your config file. Totally open to a documentation PR that explains this.
